### PR TITLE
fix(freestyle-writer) replace Optional chaining to be compatible with ui5/cli v2

### PR DESCRIPTION
**Pull request by devinea**
fix https://github.com/SAP/open-ux-tools/issues/882 

- Add additional tests for build script 
- Replace Optional chaining

should be reverted as part of https://github.com/SAP/open-ux-tools/issues/904

### CODE REVIEW

**Review by IainSAP — APPROVED** [review]
Thx @devinea . Hopefully we can revert soon... 😄

File: packages/fiori-freestyle-writer/templates/listdetail/add/webapp/controller/List.controller.ts [hunk 1]
```diff
@@ -125,7 +125,10 @@ export default class List extends BaseController {
      * and group settings and refreshes the list binding.
      */
     public onRefresh() {
-        this.list.getBinding("items")?.refresh(false);
+        const items = this.list.getBinding("items");
+        if (items){
```
Comment (ullasholla) [inline packages/fiori-freestyle-writer/templates/listdetail/add/webapp/controller/List.controller.ts:129]:
```suggestion
        if (items) {
```
Make it consistent with the rest?

File: packages/fiori-freestyle-writer/test/__snapshots__/listdetail.test.ts.snap [hunk 2]
```diff
@@ -6028,7 +6034,10 @@ export default class List extends BaseController {
      * and group settings and refreshes the list binding.
      */
     public onRefresh() {
-        this.list.getBinding(\\"items\\")?.refresh(false);
+        const items = this.list.getBinding(\\"items\\");
+        if (items){
```
Comment (ullasholla) [inline packages/fiori-freestyle-writer/test/__snapshots__/listdetail.test.ts.snap:6038]:
```suggestion
        if (items) {
```
For consistent formatting

File: packages/fiori-freestyle-writer/templates/listdetail/add/webapp/controller/List.controller.ts (hunk 1 above)
Comment (devinea) [reply packages/fiori-freestyle-writer/templates/listdetail/add/webapp/controller/List.controller.ts]: Done in https://github.com/SAP/open-ux-tools/pull/928/commits/1e8135ab0a32fc14a34815b4dfa1ab877afe4eb3

**Review by IainSAP — APPROVED** [review]
Thx @devinea

**Comment by devinea** [comment]
Thanks @IainSAP @ullasholla

**Pull request merged**